edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/CbsNode.cs

```diff
@@ -289,23 +306,22 @@ Task SendResponseAsync(AmqpMessage requestMessage, AmqpResponseStatusCode status
             }
         }
 
-        /// <summary>
-        /// This type is deliberately mutable because of the use case.
-        /// </summary>
-        class CredentialsInfo
+        internal class ClientToken
         {
-            public CredentialsInfo(IClientCredentials clientCredentials)
+            public ClientToken(string deviceId, Option<string> moduleId, string token)
             {
-                this.ClientCredentials = clientCredentials;
-                this.IsAuthenticated = false;
-                this.AsyncLock = new AsyncLock();
+                this.DeviceId = deviceId;
+                this.ModuleId = moduleId;
+                this.Token = new AtomicReference<string>(token);
```

Comment:
Why is this an AtomicRef?

Reply:
forgot to clean this up as well when I split up the struct =(

---
In reply to: [489205083](https://github.com/Azure/iotedge/pull/3552#discussion_r489205083) [](ancestors = 489205083)